File: tools/tailor_cv_ts/src/pdf.ts
import { execSync } from "child_process";
import { logger } from "./logger";

export function convertToPdf(mdPath: string, pdfPath: string): void {
  logger.info(`Converting ${mdPath} → ${pdfPath} via pandoc`);
  try {
    execSync(
      `pandoc -V papersize=a4 -V geometry:margin=1.5cm "${mdPath}" -o "${pdfPath}"`,
      { stdio: "pipe" }
    );
  } catch (err: unknown) {
    const msg = err instanceof Error ? err.message : String(err);
    throw new Error(`pandoc failed:\n${msg}`);
  }
}

File: tools/tailor_cv_py/src/tailor_cv/pdf.py
import subprocess

from loguru import logger


def convert_to_pdf(md_path: str, pdf_path: str) -> None:
    logger.info(f"Converting {md_path} → {pdf_path} via pandoc")
    result = subprocess.run(
        [
            "pandoc",
            "-f",
            "markdown",
            "-t",
            "pdf",
            "-V",
            "geometry:margin=1.5cm",
            md_path,
            "-o",
            pdf_path,
        ],
        capture_output=True,
        text=True,
    )
    if result.returncode != 0:
        raise RuntimeError(f"pandoc failed:\n{result.stderr}")
